zilliqa/src/pool.rs

```diff
@@ -1,33 +1,83 @@
+use crate::{
+    crypto::Hash,
+    state::Address,
+    transaction::{SignedTransaction, VerifiedTransaction},
+};
 use std::{
     cmp::Ordering,
     collections::{BTreeMap, BinaryHeap},
 };
+use tracing::*;
+
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord)]
+struct NoncedTxIndex {
+    pub address: Address,
+    pub nonce: u64,
+}
+
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord)]
+struct IntershardTxIndex {
+    pub source_shard: u64,
+    pub link_nonce: u64,
```

Comment:
Does `link_nonce` refer to the `bridge_nonce` as it's called elsewhere in the code?

Reply:
The main reason for keeping this nonce in the intershard case was to retain compatibility with UCCB and use the gateway contract for requesting cross-shard calls (the gateway contract injects the nonce in the encoded message structure), so calling it `uccb_nonce` makes sense to me